app/master/cluster_master.py

```diff
@@ -213,6 +213,7 @@ def handle_request_for_new_build(self, build_params):
         if build_request.is_valid():
             build = Build(build_request)
             self._all_builds_by_id[build.build_id()] = build
+            build.generate_project_type()
```

Comment:
Oh, I see your point, particularly in this case where there are some system level heavy lifting going on, like generating symlinks.  :+1:  on leaving it out